src/Core/Currency/CurrencyDataProviderInterface.php

```diff
@@ -47,14 +47,21 @@ interface CurrencyDataProviderInterface
     public function getCurrencies($object = false, $active = true, $group_by = false);
 
     /**
-     * Return raw currencies data from the database.
+     * Return raw currencies data from the database (not deleted + active currencies).
      *
      * @param bool $currentShopOnly If true returns only currencies associated to current shop
      *
-     * @return array[] Installed currencies
+     * @return array[] Available currencies
      */
     public function findAll($currentShopOnly = true);
 
+    /**
+     * Return raw currencies data from the database (regardless of their active or deleted status).
+     *
+     * @return array[] Currencies in database
+     */
+    public function findAllInDatabase();
```

Comment:
isn't a BC if the interface has already been implemented?

Reply:
Sadly yes, we discussed it with @eternoendless and couldn't find a clean way to fix this without BC break We could have kept the initial interface and modify their behaviour (first thought) but then the interfaces didn't make sense any more
Although this interface should be internal, there are very few chances of it being used outside of the core so the BC is "contained"